src/EarnerManager.sol

```diff
@@ -0,0 +1,258 @@
+            // NOTE: The `isAdmin` check in `_setDetails` will make this costly to re-set details for multiple accounts
+            //       that have already been set by the same admin, due to the redundant queries to the registrar.
+            //       Consider transient storage in `isAdmin` to memoize admins.
+            _setDetails(accounts_[index_], statuses_[index_], feeRates_[index_]);
+        }
+    }
+
+    /* ============ Temporary Admin Migration ============ */
+
+    /// @inheritdoc IEarnerManager
+    function migrate(address migrator_) external {
+        if (msg.sender != migrationAdmin) revert UnauthorizedMigration();
+
+        _migrate(migrator_);
+    }
+
+    /* ============ View/Pure Functions ============ */
+
+    /// @inheritdoc IEarnerManager
+    function earnerStatusFor(address account_) external view returns (bool status_) {
+        return earnersListsIgnored() || isInRegistrarEarnersList(account_) || isInAdministratedEarnersList(account_);
+    }
+
+    /// @inheritdoc IEarnerManager
+    function earnerStatusesFor(address[] calldata accounts_) external view returns (bool[] memory statuses_) {
+        statuses_ = new bool[](accounts_.length);
+
+        bool earnersListsIgnored_ = earnersListsIgnored();
+
+        for (uint256 index_; index_ < accounts_.length; ++index_) {
+            if (earnersListsIgnored_) {
+                statuses_[index_] = true;
+                continue;
+            }
+
+            address account_ = accounts_[index_];
+
+            if (isInRegistrarEarnersList(account_)) {
+                statuses_[index_] = true;
+                continue;
+            }
+
+            statuses_[index_] = isInAdministratedEarnersList(account_);
+        }
+    }
+
+    /// @inheritdoc IEarnerManager
+    function earnersListsIgnored() public view returns (bool isIgnored_) {
+        return IRegistrarLike(registrar).get(EARNERS_LIST_IGNORED_KEY) != bytes32(0);
+    }
+
+    /// @inheritdoc IEarnerManager
+    function isInRegistrarEarnersList(address account_) public view returns (bool isInList_) {
+        return IRegistrarLike(registrar).listContains(EARNERS_LIST_NAME, account_);
+    }
+
+    /// @inheritdoc IEarnerManager
+    function isInAdministratedEarnersList(address account_) public view returns (bool isInList_) {
+        return _isValidAdmin(_earnerDetails[account_].admin);
+    }
+
+    /// @inheritdoc IEarnerManager
+    function getEarnerDetails(address account_) external view returns (bool status_, uint16 feeRate_, address admin_) {
+        if (earnersListsIgnored() || isInRegistrarEarnersList(account_)) return (true, 0, address(0));
+
+        EarnerDetails storage details_ = _earnerDetails[account_];
+
+        // NOTE: Not using `isInAdministratedEarnersList(account_)` here to avoid redundant storage reads.
+        return _isValidAdmin(details_.admin) ? (true, details_.feeRate, details_.admin) : (false, 0, address(0));
```

Comment:
Does this mean that if an admin is removed from the registrar, `_isValidAdmin(details_.admin)` will return `false` and now anyone can call `stopEarningFor()` on Wrapped M?

Reply:
Yes. If the only reason an account is earning is because of an admin, and that admin is not longer approved, the that account is no longer approved instantly.